pkg/apis/kops/channel.go

```diff
@@ -259,13 +262,16 @@ const (
 )
 
 // FindImage returns the image for the cloudprovider, or nil if none found
-func (c *Channel) FindImage(provider CloudProviderID, kubernetesVersion semver.Version) *ChannelImageSpec {
+func (c *Channel) FindImage(provider CloudProviderID, kubernetesVersion semver.Version, architecture architectures.Architecture) *ChannelImageSpec {
 	var matches []*ChannelImageSpec
 
 	for _, image := range c.Spec.Images {
 		if image.ProviderID != string(provider) {
 			continue
 		}
+		if image.ArchitectureID != string(architecture) {
+			continue
+		}
```

Comment:
I'm wondering if we shouldn't `continue` if `image.ArchitectureID != ""`. I dont know the workflows that users have for hosting their own channels files but if they dont copy the associated architecture additions in this PR when upgrading to this version of kops then they'll no longer find any images. Perhaps a release note mentioning the change for offline installations would be sufficient.

Reply:
I think for now we should go with the release note. We can see if later we need to improve.